plugins/outputs/mqtt/mqtt.go

```diff
@@ -27,22 +29,26 @@ const (
 )
 
 type MQTT struct {
-	Servers     []string `toml:"servers"`
-	Username    string
-	Password    string
-	Database    string
-	Timeout     config.Duration
-	TopicPrefix string
-	QoS         int    `toml:"qos"`
-	ClientID    string `toml:"client_id"`
+	// General connection info
+	Servers  []string `toml:"servers"`
+	Username string
+	Password string
+	Database string
+	Timeout  config.Duration
+	ClientID string `toml:"client_id"`
 	tls.ClientConfig
-	BatchMessage bool            `toml:"batch"`
-	Retain       bool            `toml:"retain"`
-	KeepAlive    int64           `toml:"keep_alive"`
-	Log          telegraf.Logger `toml:"-"`
 
-	client paho.Client
-	opts   *paho.ClientOptions
+	// Publish info
+	TopicPrefix  string `toml:"topic_prefix"`
+	QoS          int    `toml:"qos"`
+	BatchMessage bool   `toml:"batch"`
+	Retain       bool   `toml:"retain"`
+	KeepAlive    int64  `toml:"keep_alive"`
```

Comment:
You cast to `uint16` later, so why not directly make this option `uint16`? This would help finding config issues if that value is too large...

Reply:
Yup, will change.